Website/index.js

```diff
@@ -0,0 +1,472 @@
+    var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+    $(week).each(function (i, v) {
+        if (counter < 5) {
+        var dd = new Date(v);
+        var first_date = new Date(Object.keys(jsonData["dates"])[0])
+        var last_date = new Date(Object.keys(jsonData["dates"])[Object.keys(jsonData["dates"]).length - 1]);
+        
+        var flagDuty = "";
+
+        if (flagData[v] != null) {
+
+            if (String($("#name").val()) == flagData[v]["singapore_flag"]) {
+                flagDuty = '<p><span class="fw-bold"></span><span class="badge bg-warning mb-1">Flag Duty</span><br><span class="fw-bold">Duty: </span><span id="">Singapore Flag</span></p>';
+            } else if (String($("#name").val()) == flagData[v]["school_flag"]) {
+                flagDuty = '<p><span class="fw-bold"></span><span class="badge bg-warning mb-1">Flag Duty</span><br><span class="fw-bold">Duty: </span><span id="">School Flag</span></p>';
+            }
+
+        }
+        if (dd.getDay() == 6 || dd.getDay() == 0 || jsonData["dates"][v] == 11 || dd < first_date || dd > last_date) {
+            var b = new Date(v);
+            var a = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4><p><span class="fw-bold"></span>No data is available for this day</span></p></div>'
+            $(a).appendTo("#personalised");
+        } else {
+            $("#weekend").hide();
+            var b = new Date(v);
+            var a = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4>' + flagDuty + '<p><span class="fw-bold"></span>' + recess_badge[nameData[String($("#name").val())][String(jsonData["dates"][v])]["recess"]] + '<br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())][String(jsonData["dates"][v])]["duty"]] + '</span></p></div>'
+            var c = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4>' + flagDuty + '<p><span class="fw-bold"></span><span class="badge bg-warning mb-1">Morning Duty</span><br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())]["morning_duty"]["duty"]] + '</span></p><p><span class="fw-bold"></span>' + recess_badge[nameData[String($("#name").val())][String(jsonData["dates"][v])]["recess"]] + '<br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())][String(jsonData["dates"][v])]["duty"]] + '</span></p></div>'                    
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Davianlols6/SC-Duty-List/pullRequest?prid=8386818)
```suggestion
            var c = "<div id=\"" + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4>' + flagDuty + '<p><span class="fw-bold"></span><span class="badge bg-warning mb-1">Morning Duty</span><br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())]["morning_duty"]["duty"]] + '</span></p><p><span class="fw-bold"></span>' + recess_badge[nameData[String($("#name").val())][String(jsonData["dates"][v])]["recess"]] + '<br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())][String(jsonData["dates"][v])]["duty"]] + '</span></p></div>'                    
```